strawberryfields/configuration.py

```diff
@@ -0,0 +1,330 @@
+# Copyright 2019-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+This module contains functions used to load, store, save, and modify
+configuration options for Strawberry Fields.
+"""
+import logging as log
+import os
+
+import toml
+from appdirs import user_config_dir
+
+log.getLogger()
+
+DEFAULT_CONFIG_SPEC = {
+    "api": {
+        "authentication_token": (str, ""),
+        "hostname": (str, "platform.strawberryfields.ai"),
+        "use_ssl": (bool, True),
+        "port": (int, 443),
+    }
+}
+
+
+class ConfigurationError(Exception):
+    """Exception used for configuration errors"""
+
+
+def load_config(filename="config.toml", **kwargs):
+    """Load configuration from keyword arguments, configuration file or
+    environment variables.
+
+    .. note::
+
+        The configuration dictionary will be created based on the following
+        (order defines the importance, going from most important to least
+        important):
+
+        1. keyword arguments passed to ``load_config``
+        2. data contained in environmental variables (if any)
+        3. data contained in a configuration file (if exists)
+
+    Kwargs:
```

Comment:
`Keyword Args` seem to be more commonly used in SF, while `Kwargs` is only used in this file (`configuration.py`).

```suggestion
    Keyword Args:
```